redisenterprise/datadog_checks/redisenterprise/check.py

```diff
@@ -0,0 +1,319 @@
+        self.last_event_timestamp_seen = datetime.utcnow() - timedelta(0, 120)
+
+    def _timestamp(self, date):
+        """ Allows us to return an epoch time stamp if we use python2 or python3 """
+        if sys.version_info[0] < 3 or sys.version_info[1] < 4:
+            import time
+
+            return int(time.mktime(date.timetuple()))
+        else:
+            return int(date.timestamp())
+
+    def check(self, instance):
+        host = self.instance.get('host')
+        user = self.instance.get('username')
+        password = self.instance.get('password')
+        timeout = self.instance.get('timeout')
+        port = self.instance.get('port', 9443)
+        event_limit = self.instance.get('event_limit', 100)
+        is_mock = self.instance.get('is_mock', False)
+        service_check_tags = self.instance.get('tags', [])
+
+        if not host or not user or not password:
+            raise ConfigurationError('Configuration error, please fix redisenterprise.yaml')
+
+        try:
+
+            # noop if we are not the cluser master
+            if self._check_follower(host, port, user, password, timeout, is_mock):
+                self.last_timestamp_seen = datetime.utcnow()
+                pass
+
+            # add the cluster FQDN to the tags
+            fqdn = self._get_fqdn(host, port, user, password, service_check_tags)
+            service_check_tags.append('redis_cluster:{}'.format(fqdn))
+
+            # collect the license data
+            fqdn = self._get_license(host, port, user, password, service_check_tags)
+
+            # collect the node data
+            self._get_nodes(host, port, user, password, service_check_tags)
+
+            # grab the DBD ID to name mapping
+            bdb_dict = self._get_bdb_dict(host, port, user, password, service_check_tags)
+            self._get_bdb_stats(host, port, user, password, bdb_dict, service_check_tags)
+            self._shard_usage(bdb_dict, service_check_tags, host)
+
+            # collect the events from the API - we set the timeout higher here
+            self._get_events(host, port, user, password, bdb_dict, service_check_tags, event_limit)
+
+            self.service_check(
+                'redisenterprise.running',
+                self._get_version(host, port, user, password, service_check_tags),
+                tags=service_check_tags,
+            )
+        except Exception as e:
+            self.service_check('redisenterprise.running', self.CRITICAL, message=str(e), tags=service_check_tags)
+
+        pass
+
+    def _check_follower(self, host, port, user, password, timeout, is_mock):
+        """ The RedisEnterprise returns a 307 if a node is a cluster follower (not leader) """
+        if is_mock:
+            return False
+        headers_sent = {'Content-Type': 'application/json'}
+        url = 'https://{}:{}/v1/cluster'.format(host, port)
+        r = self.http.get(url, extra_headers=headers_sent, auth=HTTPBasicAuth(user, password))
```

Comment:
Removing the basic auth section causes the get to fail

```
r = self.http.get(url, extra_headers=headers_sent)
```
logs:
```
[2021-07-10T16:06:17.179Z] "GET /v1/cluster HTTP/1.1" 401 - 0 0 3 3 "10.0.77.4" "Datadog Agent/7.26.0" "60f7ac6a-c09e-4f2d-9487-0a63999163ae" "localhost:9443" "127.0.0.1:9080"

```

So I would posit that needs to remain in the configuration as the end user will not be aware of the basic auth distinctions.

Also I tried setting the extra_headers in HTTP_CONFIG_REMAPPER as well, but they were not getting picked up.  Is there somewhere else?

@coignetp

Reply:
The reason is because the default option for the user is `username` and not `user` https://github.com/DataDog/integrations-extras/pull/849/files#diff-73c04ac247c1ee8a0ebbc067fdfb32b4f6c95f4c96d1395dc37456cd8f6c15a5R97
You can check the code in https://github.com/DataDog/integrations-core/blob/820e7ea3d54012437cf3d355e4c53a885ad4fae5/datadog_checks_base/datadog_checks/base/utils/http.py#L570